### PR TITLE
Handle new vector opcodes

### PR DESCRIPTION
**Pull request by gita-omr**
Depends on https://github.com/eclipse/omr/pull/6942

### CODE REVIEW

**Comment by gita-omr** [comment]
Added commit to handle compressExpandOp() intrinsic.

**Comment by Akira1Saitoh** [comment]
It seems that we need to recognize `VectorSupport.broadcastInt` to vectorize lanewise shift operations with immediate shift amount.
https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/e336ea0f0702e0a790ad0f52658b093fc1a86b8a/src/jdk.incubator.vector/share/classes/jdk/incubator/vector/IntVector.java#L1008-L1020

**Comment by gita-omr** [comment]
> It seems that we need to recognize `VectorSupport.broadcastInt` to vectorize lanewise shift operations with immediate shift amount. https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/e336ea0f0702e0a790ad0f52658b093fc1a86b8a/src/jdk.incubator.vector/share/classes/jdk/incubator/vector/IntVector.java#L1008-L1020

Thanks! Trying to figure out how it's different from `fromBitsCoerced` intrinsic for which we generate `vsplats`.

**Comment by gita-omr** [comment]
`fromBitsCoerced` takes` long` for all types and that `long` is a bit pattern that needs to be converted to each type. `broadcastInt` is for non-FP types only and takes ` int` as a parameter. We will generate `vsplats` for it as well, just without conversion.

**Comment by gita-omr** [comment]
Actually, `broadcastInt`  seems to be doing a shift by the same constant of each vector element (unary operation). @BradleyWood @Akira1Saitoh do you think it's better to have a new opcode (something like `vshrConst`) or convert it into `vsplats` of the constant and then `vshr` (for example) ?

**Comment by Akira1Saitoh** [comment]
@gita-omr Either way works for AArch64. I have already drafted vector shift evaluators with the assumption that `vsplats` of the constant comes as the second child of vector shift node for `broadcastInt` case. 
https://github.com/Akira1Saitoh/omr/commit/c0a7e2fbbb48f5ed1dd9fc1b565514350e409437
AArch64 has dedicated instructions for vector shifts with immediate value as well as vector shifts with the value in the register.
https://developer.arm.com/documentation/ddi0596/2020-12/SIMD-FP-Instructions/SHL--Shift-Left--immediate--
https://developer.arm.com/documentation/ddi0596/2020-12/SIMD-FP-Instructions/SSHL--Signed-Shift-Left--register--

**Comment by gita-omr** [comment]
> @gita-omr Either way works for AArch64. I have already drafted vector shift evaluators with the assumption that `vsplats` of the constant comes as the second child of vector shift node for `broadcastInt` case.

Thanks @Akira1Saitoh ! What if vsplats <const> is put in a register, for example by PRE. Then, we will not be able to use vector shift with immediate instruction.

**Comment by Akira1Saitoh** [comment]
@gita-omr Agree. If we have dedicated opcodes, the evaluator will be able to use appropriate instructions without getting affected by other optimizations.

**Comment by Akira1Saitoh** [comment]
@gita-omr Looks like the same applies to rotate operations.
> It seems that we need to recognize `VectorSupport.broadcastInt` to vectorize lanewise shift operations with immediate shift amount. https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/e336ea0f0702e0a790ad0f52658b093fc1a86b8a/src/jdk.incubator.vector/share/classes/jdk/incubator/vector/IntVector.java#L1008-L1020

**Comment by gita-omr** [comment]
> @gita-omr Looks like the same applies to rotate operations

Thanks @Akira1Saitoh . I will add those too.

File: runtime/compiler/optimizer/VectorAPIExpansion.hpp [hunk 1]
```diff
@@ -1322,6 +1456,6 @@ class TR_VectorAPIExpansion : public TR::Optimization
    *   \return
    *      Transformed node
    */
-   static TR::Node *transformNary(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node, TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes, handlerMode mode, TR::ILOpCodes scalarOpCode, TR::ILOpCodes vectorOpCode, int32_t firstOperand, int32_t numOperands, vapiOpCodeType opCodeType);
+   static TR::Node *transformNary(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node, TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes, handlerMode mode, TR::ILOpCodes scalarOpCode, TR::ILOpCodes vectorOpCode, int32_t firstOperand, int32_t numOperands, vapiOpCodeType opCodeType, bool transformRORtoROL);
```
Comment (knn-k) [inline runtime/compiler/optimizer/VectorAPIExpansion.hpp:1459]: Please add the description of the new parameter in the comment above.

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 2]
```diff
@@ -2178,6 +2320,12 @@ TR::ILOpCodes TR_VectorAPIExpansion::ILOpcodeFromVectorAPIOpcode(TR::Compilation
             return TR::BadILOp;
          }
       }
+
+
+
+
+
+
```
Comment (knn-k) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:2328]: What is the purpose of these empty lines?

**Comment by gita-omr** [comment]
Addressed comments above.

**Comment by knn-k** [comment]
Jenkins test sanity all jdk17 depends eclipse/omr#6942

**Review by knn-k — APPROVED** [review]
Waiting for https://github.com/eclipse/omr/pull/6942 to be merged.

**Comment by 0xdaryl** [comment]
> Waiting for https://github.com/eclipse/omr/pull/6942 to be merged.

Because 6942 is needed for an OpenJ9 feature, I am holding off merging it until the OMR Acceptance build is successful.  Even though the risk is low, I don't want to inject any unnecessary instability into that build because it has been nearly 2 weeks since the last promotion.

**Pull request merged**